tests/php/test_class-wc-connect-shipping-method.php

```diff
@@ -99,4 +99,64 @@ public function test_is_valid_package_destination( $package, $expected ) {
 
 	}
 
+	/**
+	 * @covers WC_Connect_Shipping_Method::calculate_shipping
+	 */
+	public function test_invalid_destination_calculate_shipping() {
+
+		$loader = new WC_Connect_Loader();
+		$loader->load_dependencies();
+
+		$shipping_method = $this->getMockBuilder( 'WC_Connect_Shipping_Method' )
+			->disableOriginalConstructor()
+			->setMethods( array( 'is_valid_package_destination', 'log' ) )
+			->getMock();
+
+		$shipping_method->expects( $this->any() )
+			->method( 'is_valid_package_destination' )
+			->will( $this->returnValue( false ) );
+
+		$shipping_method->expects( $this->once() )
+			->method( 'log' )
+			->with(
+				$this->stringContains( 'Package destination failed validation.' ),
+				$this->anything()
+			);
+
+		$shipping_method->calculate_shipping();
+
+	}
+
+	/**
+	 * @covers WC_Connect_Shipping_Method::calculate_shipping
+	 */
+	public function test_invalid_service_settings_calculate_shipping() {
+
+		$loader = new WC_Connect_Loader();
+		$loader->load_dependencies();
+
+		$shipping_method = $this->getMockBuilder( 'WC_Connect_Shipping_Method' )
+			->setConstructorArgs( array( 1 ) )
+			->setMethods( array( 'is_valid_package_destination', 'get_service_settings', 'log' ) )
+			->getMock();
+
+		$shipping_method->expects( $this->any() )
+			->method( 'is_valid_package_destination' )
+			->will( $this->returnValue( true ) );
+
+		$shipping_method->expects( $this->any() )
+			->method( 'get_service_settings' )
+			->will( $this->returnValue( new stdClass() ) );
+
+		$shipping_method->expects( $this->once() )
+			->method( 'log' )
+			->with(
+				$this->stringContains( 'Service settings empty.' ),
+				$this->anything()
+			);
+
+		$shipping_method->calculate_shipping();
+
+	}
+
 }
```

Comment:
I was trying to test the code I added without going overboard and getting full coverage for unrelated changes. Though yes, eventually this should be 100% covered.